Maple2.Server.Game/PacketHandlers/SkillHandler.cs

```diff
@@ -128,6 +128,33 @@ private enum SubCommand : byte {
             }
         }
 
+        int spiritCost = record.GetSpCost();
+        //TODO: Proper invoke cost
+        /*
+        InvokeStatValue invokeStat = Stats.GetSkillStats(skillCast.SkillId, skillCast.GetSkillGroups(), InvokeEffectType.ReduceSpiritCost);
+        spiritCost = Math.Max(0, (int) (-invokeStat.Value + (1 - invokeStat.Rate) * spiritCost));
+        */
+        if (spiritCost > 0) {
+            if (session.Player.Stats[BasicAttribute.Spirit].Current < spiritCost) {
+                Logger.Error("Not enough spirit to cast skill: {SkillId},{Level}", skillId, level);
+                return;
+            }
+        }
+
+        int staminaCost = record.GetStaCost();
+        if (staminaCost > 0) {
+            if (session.Player.Stats[BasicAttribute.Stamina].Current < staminaCost) {
+                Logger.Error("Not enough stamina to cast skill: {SkillId},{Level}", skillId, level);
+                return;
+            }
+        }
+
+        session.Player.ConsumeSp(spiritCost);
+        session.Player.ConsumeStamina(staminaCost);
+        session.Field.Broadcast(StatsPacket.Update(session.Player, [BasicAttribute.Spirit, BasicAttribute.Stamina]));
+
+        session.ConditionUpdate(ConditionType.skill, 1, codeLong: skillId);
```

Comment:
`session.ConditionUpdate(ConditionType.skill, codeLong: skillId);`

You'll also need to handle this in Server.Game/Util/ConditionUtil